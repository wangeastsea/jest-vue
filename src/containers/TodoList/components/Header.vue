<template>
    <div class="header">
        <div class="header-content">
            TodoList
            <input
                class="header-input"
                data-test='header-input'
                v-model="inputValue"
                @keyup.enter="addTodoItem"
                >
        </div>
    </div>
</template>
<script>
export default {
  name: 'Header',
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>
<style lang="stylus">
    .header {
        background: #333;
        line-height: 60px;
    }
    .header-content {
        width: 600px;
        margin: 0 auto;
        color: #FFF;
        font-size: 16px;
    }
    .header-input {
        float: right;
        width: 360px;
        line-height: 24px;
        margin-top: 12px;
        outline: none;
        color: #333;
    }
</style>
