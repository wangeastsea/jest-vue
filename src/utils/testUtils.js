// 统一封装函数
export const findTestWrapper = (wrapper, tag) => {
  return wrapper.findAll(`[data-test="${tag}"]`)
}
