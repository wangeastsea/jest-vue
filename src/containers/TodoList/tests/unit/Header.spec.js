import { shallowMount } from '@vue/test-utils'
import Header from '@/containers/TodoList/components/Header.vue'
import { findTestWrapper } from '@/utils/testUtils'

describe('Header 组件', () => {
  it('包含input框', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    expect(input.exists()).toBe(true)
  })

  it('input框的初始值为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })

  it('input发生变化，数据应该跟着变化', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('dong hai')
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('dong hai')
  })

  it('input发生变化，无内容时， 无反应', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  it('input发生变化，有内容时，向外触发事件', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('234')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
  })

  it('input发生变化，有内容时，向外触发事件,并清空input框', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('234')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    expect(wrapper.vm.$data.inputValue).toBe('')
  })

  // 快照功能
  it('样式发生改变，给出提示', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
  })
})
