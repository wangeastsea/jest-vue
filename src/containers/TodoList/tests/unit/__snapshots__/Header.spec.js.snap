// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header样式发生改变，给出提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input data-test="input" class="header-input"></div>
</div>
`;
